internal/graph/graph.go

```diff
@@ -431,12 +433,22 @@ func NewLayeredCheckResolver(
 	localResolverOpts []LocalCheckerOption,
 	cacheEnabled bool,
 	cachedResolverOpts []CachedCheckResolverOpt,
+	throttlingEnabled bool,
```

Comment:
Agree, and I will add that this function needs to be moved to its own file with its own unit tests, and merge this code with this huge blob https://github.com/openfga/openfga/blob/4788f8da2529e1364378b33a49c9eeff961ccdbb/pkg/server/server.go#L426-L489

I'm actually concerned that this code is split, because can we be 100% sure that we haven't messed up the order?

Reply:
Added comments to address this in seperate PR - https://github.com/openfga/openfga/pull/1571#issuecomment-2093143714